lib/kvdb/indent.c

```diff
@@ -0,0 +1,158 @@
+
+#include "build_config.h"
+
+#include <string.h>
+
+#include <bsd/libutil.h>
+#include <bsd/string.h>
+
+#include <hse/hse.h>
+#include <hse/experimental.h>
+#include <hse/version.h>
+
+#include <hse/config/config.h>
+#include <hse/ikvdb/hse_gparams.h>
+#include <hse/ikvdb/ikvdb.h>
+#include <hse/ikvdb/kvdb_cparams.h>
+#include <hse/ikvdb/kvdb_ctxn.h>
+#include <hse/ikvdb/kvdb_home.h>
+#include <hse/ikvdb/kvdb_perfc.h>
+#include <hse/ikvdb/kvdb_rparams.h>
+#include <hse/ikvdb/kvs.h>
+#include <hse/ikvdb/kvs_cparams.h>
+#include <hse/ikvdb/kvs_rparams.h>
+#include <hse/ikvdb/limits.h>
+#include <hse/logging/logging.h>
+#include <hse/mpool/mpool.h>
+#include <hse/pidfile/pidfile.h>
+#include <hse/rest/headers.h>
+#include <hse/rest/params.h>
+#include <hse/rest/request.h>
+#include <hse/rest/response.h>
+#include <hse/rest/server.h>
+#include <hse/rest/status.h>
+#include <hse/util/err_ctx.h>
+#include <hse/util/event_counter.h>
+#include <hse/util/mutex.h>
+#include <hse/util/platform.h>
+#include <hse/util/vlb.h>
+#include <hse/util/compiler.h>
+
+#define HSE_KVS_PUT_MASK       1
+#define HSE_KVS_PUT_VCOMP_MASK 2
+
+char *
+some_long_function_name(void *);
+
+bool long_variable_boolean;
+int long_variable_integer;
+void * long_variable_pointer;
+struct foobar_foobar;
+void * const * yoyo;
+
+
+struct foo {
+    char foobar_foobar_foobar[1234];
+    int * foo;
+    unsigned long ** bar;
+};
+
+
+// Func Decls
+int
+function1(unsigned int * arg1);
+
+int
+function2(unsigned int * arg1, unsigned int ** arg2, unsigned int *** arg3, unsigned int arg4);
+
+int
+function3(
+    unsigned int * arg1,
+    unsigned int ** arg2,
+    unsigned int *** arg3,
+    unsigned int arg4,
+    unsigned int arg5)
+{
+    *arg1 = 0;
+    **arg2 = 1;
+    return 0;
+}
+
+
+// Func Decls w/ Attributes
+
+void HSE_NORETURN
+fn_attr1(void);
+
+HSE_NORETURN void
+fn_attr2(void);
+
+void HSE_PRINTF(1, 2)
+fn_attr3(char * fmt);
+
+void HSE_NORETURN
+fn_attr4(unsigned int arg1, unsigned int arg2, unsigned int arg3, struct foobar_foobar * foobar);
+
+HSE_NORETURN void
+fn_attr5(unsigned int arg1, unsigned int arg2, unsigned int arg3, struct foobar_foobar * foobar);
+
+void HSE_PRINTF(1, 2)
+fn_attr6(unsigned int arg1, unsigned int arg2, unsigned int arg3, struct foobar_foobar * foobar);
+
+
+hse_err_t
+long_if_expr(int flags)
+{
+    if (long_variable_boolean || long_variable_integer > 0 || long_variable_pointer != NULL ||
+        (flags & HSE_KVS_PUT_VCOMP_MASK) == HSE_KVS_PUT_VCOMP_MASK)
```

Comment:
It is very unfortunate that clang-format can't differentiate between these types of continuation lines vs other.

Reply:
Agree